lib/lotus/model/migration.rb

```diff
@@ -0,0 +1,163 @@
+    #
+    #   * add_column
+    #   * add_index
+    #   * create_view
+    #   * drop_column
+    #   * drop_index
+    #   * drop_table
+    #   * drop_view
+    #   * rename_table
+    #   * rename_column
+    #   * set_column_default
+    #   * set_column_type
+    #
+    # For more details, please consult documentation
+    # at http://sequel.jeremyevans.net/rdoc/files/doc/schema_modification_rdoc.html
+    #
+    # @xample
+    #   require 'lotus/model/migration'
+    #
+    #   class MyMigration < Lotus::Model::Migration
+    #     def up
+    #       create_table(:authors) do
+    #         primary_key :id
+    #         String :name
+    #       end
+    #     end
+    #
+    #     def down
+    #       drop_table(:authors)
+    #     end
+    #   end
+    #
+    #   # to migrate
+    #   MyMigration.apply(DB, :up)
+    #
+    #   # to rollback
+    #   MyMigration.apply(DB, :down)
+    #
+    # @since x.x.x
+    # @see Sequel::Migration
+    Migration = ::Sequel::Migration
+
+
+    # The +Migrator+ class performs migrations based on migration files in
+    # db/migrations directory. The migration files should be named using the
+    # following pattern:
+    #
+    #   <yyyymmddhhmmss>_<title>.rb
+    #
+    # For example:
+    #
+    #   20150122124517_create_users.rb
+    #
+    # The migration file must extend +Lotus::Model::Migration+ and
+    # implement +up+ and +down+ methods.
+    # @see Lotus::Model::Migration
+    #
+    # To apply a migrator, the +apply+ must be invoked with the
+    # SQL adapter config instance, and optional target version.
+    # If no current version is supplied, it is read from the database.
+    # The migrator automatically create schema_migration table to
+    # keep track of current migration version. If no migration is
+    # stored in the database, the version is considered to be 0.
+    # if no target version is specified, the database is migrated
+    # to the latest version available in db/migration directory.
+    #
+    # @example Migrate all migrations in db/migrations to latest
+    #   require 'lotus/model/migration'
+    #   adapter_cfg = Lotus::Model::Config::Adapter.new(type: :sql, uri: DATABASE_URI)
+    #   Lotus::Model::Migrator.new(adapter_cfg).run
+    #
+    # @example To use shortcut +apply+ method to invoke +run+ instance method
+    #   require 'lotus/model/migration'
+    #   adapter_cfg = Lotus::Model::Config::Adapter.new(type: :sql, uri: DATABASE_URI)
+    #   Lotus::Model::Migrator.apply(adapter_cfg)
+    #
+    # @since x.x.x
+    #
+    # @see Sequel::Migration
+    # @see Sequel::TimestampMigrator
+    class Migrator
+      extend Forwardable
+
+      # Run the migrator
+      #
+      # @since x.x.x
+      # @delegate Sequel::TimestampMigrator#run
+      #
+      # @see Sequel::TimestampMigrator#run
+      delegate :run => :migrator
+
+      MIGRATION_DIRECTORY = "#{Dir.pwd}/db/migrations".freeze
```

Comment:
Come to think of it, there is `Lotus::Environment`, I am going to see if I could use it